src/com/chrome/apis/Controller.js

```diff
@@ -0,0 +1,299 @@
+        }
+      */}
+    })
+  ]
+});
+
+foam.CLASS({
+  package: 'foam.u2.md',
+  name: 'DAOCreateController',
+  extends: 'foam.u2.DAOCreateController',
+
+  requires: [
+    'foam.u2.md.Button',
+    'foam.u2.md.DetailView'
+  ],
+
+  imports: [
+    'toolbar'
+  ],
+
+  // TODO(braden) : These actions are duplicated. When Actions inherit properly,
+  // these just need to set the icon, and not duplicate the code.
+  actions: [
+    {
+      name: 'cancel',
+      icon: 'clear',
+      code: function() {
+        this.stack.popMe(this);
+      }
+    },
+    {
+      name: 'save',
+      icon: 'check',
+      code: function() {
+        this.clearFocus_();
+        this.dao.put(this.data).then(function() {
+          this.stack.popMe(this);
+        }.bind(this));
+      }
+    }
+  ],
+
+  methods: [
+    function initE() {
+      this.add(this.buildDetailView());
+      //var dv = this.DetailView.create({ of: this. of });
+      //dv.data$ = this.data$;
+      //this.add(dv);
+
+      this.toolbar.title$ = this.title$;
+      this.toolbar.addLeftAction(this.buildButton(this.CANCEL));
+      this.toolbar.addRightAction(this.buildButton(this.SAVE));
+    },
+
+    function buildButton(action) {
+      return this.Button.create({
+        action: action,
+        type: 'icon',
+        data: this
+      });
+    }
+  ]
+});
+
+
+foam.CLASS({
+  package: 'foam.u2.md',
+  name: 'DAOUpdateController',
+  extends: 'foam.u2.DAOUpdateController',
+
+  requires: [
+    'foam.u2.md.Button',
+    'foam.u2.md.DetailView'
+  ],
+
+  imports: [
+    'toolbar'
+  ],
+
+  // TODO(braden) : These actions are duplicated. When Actions inherit properly,
+  // these just need to set the icon, and not duplicate the code.
+  actions: [
+    {
+      name: 'back',
+      icon: 'arrow-back',
+      isAvailable: function(dirty_) { return ! dirty_; },
+      code: function(X) {
+        X.stack.popMe(this);
+      }
+    },
+    {
+      name: 'delete',
+      icon: 'delete',
+      isAvailable: function(dirty_) { return ! dirty_; },
+      code: function(X) {
+        this.dao.remove(this.data);
+        this.back();
+      }
+    },
+    {
+      name: 'cancel',
+      icon: 'clear',
+      isAvailable: function(dirty_) { return dirty_; },
+      code: function(X) {
+        this.clearFocus_();
+        this.innerData = this.data.clone();
+      }
+    },
+    {
+      name: 'submit',
+      icon: 'check',
+      isAvailable: function(dirty_) { return dirty_; },
+      code: function(X) {
+        // TODO(braden): Handle long-running submits here. Some UI affordance
+        // for the fact that work is being done here.
+        this.clearFocus_();
+        this.dao.put(this.innerData).then(function(nu) {
+          this.data = nu;
+          this.innerData = nu.clone();
+        }.bind(this));
+      }
+    }
+  ],
+
+
+  methods: [
+    function initE() {
+      var dv = this.DetailView.create({ of: this. of });
+      dv.data$ = this.innerData$;
```

Comment:
This can be done in the create() call now.  Then you no longer need to specify 'of'.